docs/docsite/rst/aws_ec2_guide.rst

```diff
@@ -0,0 +1,523 @@
+.. _ansible_collections.amazon.aws.docsite.dynamic_inventory:
+
+
+Dynamic Inventory Plugin
+========================
+
+``aws_ec2`` dynamic inventory plugin allow users to point at data sources to compile the inventory of AWS EC2 hosts that Ansible uses to target tasks, either via the ``-i /path/to/file`` and/or ``-i 'host1, host2'`` command line parameters or from other configuration sources.
+
+Once your AWS EC2 hosts are spun up, you'll probably want to talk to them again. With a cloud setup, it's best to not maintain a static list of cloud hostnames
+in text files. Rather, the best way to handle this is to use the ``aws_ec2`` dynamic inventory plugin.
+
+The plugin will also return instances that were created outside of Ansible and allow Ansible to manage them.
+
+.. _ansible_collections.amazon.aws.docsite.using_inventory_plugin:
+
+Minimal Example
+===============
+
+To start using the ``aws_ec2`` dynamic inventory plugin with a YAML configuration source, create a file with the accepted filename schema documented for the plugin in question (a YAML configuration file that ends with aws_ec2.(yml|yaml), e.g., demo.aws_ec2.yml), then add ``plugin: amazon.aws.aws_ec2``. Use the fully qualified name if the plugin is in a collection.
+
+Fetch all hosts in us-east-1, the hostname is the public DNS if it exists, otherwise the private IP address.
+
+.. code-block:: yaml
+
+    # demo.aws_ec2.yml
+    plugin: amazon.aws.aws_ec2
+
+    # If your Ansible server is running inside the AWS environment, attach an EC2 instance role with the
+    # required AWS EC2 permissions. This way you don't have to add the access and secret key in the
+    # configuration. Ansible will automatically use the attached role to make the AWS API calls.
+    iam_role_arn: arn:aws:iam::1234567890:role/assumed-ansible
+
+    # If your Ansible server is not in the AWS then you can not attach IAM role to it so instead
+    # of that you set `aws_access_key` and `aws_secret_key`.
+    aws_access_key: <YOUR-AWS-ACCESS-KEY-HERE>
+    aws_secret_key: <YOUR-AWS-SECRET-KEY-HERE>
+
+    # Or attach the default AWS profile
+    aws_profile: default
+
+    # Or you could use Jinja2 to attach the AWS profile from the environment variable.
+    aws_profile: "{{ lookup('env', 'AWS_PROFILE') | default('dev-profile', true) }}"
```

Comment:
Authentication generally wants to be an "a xor b xor c".  As such I'd be tempted to split it out to specifically talk about authentication pieces.